### PR TITLE
Follow up synchronizations

### PR DESCRIPTION
**Pull request by rjzondervan**

### CODE REVIEW

File: lib/Service/SynchronizationService.php [hunk 1]
```diff
@@ -136,6 +137,12 @@ public function synchronize(Synchronization $synchronization, ?bool $isTest = fa
             $this->synchronizationContractMapper->update($synchronizationContract);
         }
 
+		foreach($synchronization->getFollowUps() as $followUp) {
+			var_dump($followUp);
```
Comment (bbrands02) [inline lib/Service/SynchronizationService.php:141]: var-dump

File: lib/Migration/Version1Date20241126074122.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+use Closure;
+use OCA\OpenRegister\Db\Schema;
+use OCP\DB\ISchemaWrapper;
+use OCP\DB\Types;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+
+/**
+ * FIXME Auto-generated migration step: Please modify to your needs!
```
Comment (WilcoLouwerse) [inline lib/Migration/Version1Date20241126074122.php:20]: Optional: you could add here what this migration does like so:
image

File: lib/Service/SynchronizationService.php [hunk 3]
```diff
@@ -90,6 +91,8 @@ public function synchronize(Synchronization $synchronization, ?bool $isTest = fa
 
         $objectList = $this->getAllObjectsFromSource(synchronization: $synchronization, isTest: $isTest);
 
+//		var_dump(count($objectList));
```
Comment (WilcoLouwerse) [inline lib/Service/SynchronizationService.php:94]: another var dump

File: lib/Service/SynchronizationService.php [hunk 4]
```diff
@@ -136,6 +137,11 @@ public function synchronize(Synchronization $synchronization, ?bool $isTest = fa
             $this->synchronizationContractMapper->update($synchronizationContract);
         }
 
+		foreach($synchronization->getFollowUps() as $followUp) {
```
Comment (WilcoLouwerse) [inline lib/Service/SynchronizationService.php:140]: foreach (

File: lib/Service/SynchronizationService.php [hunk 5]
```diff
@@ -246,14 +252,19 @@ public function synchronizeContract(SynchronizationContract $synchronizationCont
...
+			return $synchronizationContract;
         }
 
         // The object has changed, oke let do mappig and bla die bla
         $synchronizationContract->setOriginHash($originHash);
         $synchronizationContract->setSourceLastChanged(new DateTime());
 
+		// Check if object adheres to conditions.
+		// Take note, JsonLogic::apply() returns a range of return types, so checking it with '=== false' or '!== true' does not work properly.
+		if ($synchronization->getConditions() !== [] && !JsonLogic::apply($synchronization->getConditions(), $object)) {
```
Comment (WilcoLouwerse) [inline lib/Service/SynchronizationService.php:264]: === false

**Review by WilcoLouwerse — CHANGES_REQUESTED** [review]

File: lib/Service/SynchronizationService.php (hunk 5 above)
Comment (rjzondervan) [reply lib/Service/SynchronizationService.php:264]: Did you read the comment above this line? === false is not going to cut it.

**Review by WilcoLouwerse — APPROVED** [review]

**Pull request merged**